lib/meson.build

```diff
@@ -101,7 +101,7 @@ custom_target(
     command: [
         g_ir_compiler,
         '--shared-library',
-        libgranite.full_path(),
+        libgranite.full_path().split('/')[-1],
```

Comment:
This is not my area of expertise, but I've taken a look to see how other libraries handle this: https://github.com/search?q=language:meson+"--shared-library"&type=Code

I think that splitting the string is a bit unconventional, maybe better to use one of the libraries from above (granite_gi?) to build this argument?

Reply:
I did check roughly the same source, and I resorted to splitting the path because the [`library()`](https://mesonbuild.com/Reference-manual.html#library) rule can potentially alter the output file name in several ways (e.g. through custom `name_prefix` or `name_suffix`).

Though, checking the same search again, though, I noticed a neat way of achieving the same in [`saga-glib`](https://github.com/major-lab/saga-glib/blob/9b7b5618e259bb88e84b137d339c3520790db200/src/meson.build#L36-L37): adding the library as `input` and referring to it via `@PLAINNAME@` which does exactly what I need here without having to resort to string operations ([documentation](https://mesonbuild.com/Reference-manual.html#custom_target)).

I'll try to update the commit accordingly.

Reply:
After a bit of fight with Git, I think I managed to update it cleanly (5fa5bbbcb0075796b2667a5ee6098221ebc935ab). PTAL.

Reply:
Ping?

Reply:
The new commit looks much cleaner :+1:, but I'd prefer someone with more knowledge about the Debian packaging to take a look at this.